specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2018-04-04/policyStates.json

```diff
@@ -0,0 +1,1516 @@
+          "type": "string"
+        },
+        "@odata.context": {
+          "description": "OData context string; used by OData clients to resolve type information based on metadata.",
+          "type": "string"
+        },
+        "results": {
+          "description": "Non-compliance summary for all policy assignments.",
+          "$ref": "#/definitions/SummaryResults"
+        },
+        "policyAssignments": {
+          "description": "Policy assignments summary.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PolicyAssignmentSummary"
+          }
+        }
+      }
+    },
+    "SummaryResults": {
+      "description": "Non-compliance summary on a particular summary level.",
+      "properties": {
+        "queryResultsUri": {
+          "description": "HTTP POST URI for queryResults action on Microsoft.PolicyInsights to retrieve raw results for the non-compliance summary.",
+          "type": "string"
+        },
+        "nonCompliantResources": {
+          "description": "Number of non-compliant resources.",
+          "type": "integer",
+          "format": "int32",
+          "minimum": 0
+        },
+        "nonCompliantPolicies": {
+          "description": "Number of non-compliant policies.",
+          "type": "integer",
+          "format": "int32",
+          "minimum": 0
+        }
+      }
+    },
+    "PolicyAssignmentSummary": {
+      "description": "Policy assignment summary.",
+      "properties": {
+        "policyAssignmentId": {
+          "description": "Policy assignment ID.",
+          "type": "string"
+        },
+        "policySetDefinitionId": {
+          "description": "Policy set definition ID, if the policy assignment is for a policy set.",
+          "type": "string"
+        },
+        "results": {
+          "description": "Non-compliance summary for the policy assignment.",
+          "$ref": "#/definitions/SummaryResults"
+        },
+        "policyDefinitions": {
+          "description": "Policy definitions summary.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PolicyDefinitionSummary"
+          }
+        }
+      }
+    },
+    "PolicyDefinitionSummary": {
+      "description": "Policy definition summary.",
+      "properties": {
+        "policyDefinitionId": {
+          "description": "Policy definition ID.",
+          "type": "string"
+        },
+        "policyDefinitionReferenceId": {
+          "description": "Policy definition reference ID.",
+          "type": "string"
+        },
+        "effect": {
+          "description": "Policy effect, i.e. policy definition action.",
+          "type": "string"
+        },
+        "results": {
+          "description": "Non-compliance summary for the policy definition.",
+          "$ref": "#/definitions/SummaryResults"
+        }
+      } 
+    },
+    "QueryFailure": {
+      "description": "Error response.",
+      "properties": {
+        "error": {
```

Comment:
>error [](start = 9, length = 5)

I am  not a big fan of defining stuff inline

Reply:
I've found this easier to read since there are just two parameters and is extremely unlikely to change in the future (all of ARM is using the same). If you are feeling strong about it I can change; I just want to avoid changes from the previous specs if I can to minimize risk here.